xbmc/video/jobs/VideoLibraryRefreshingJob.cpp

```diff
@@ -74,11 +78,128 @@ bool CVideoLibraryRefreshingJob::Work(CVideoDatabase &db)
+          if (!URIUtils::IsPlugin(movie->GetDynPath()))
+          {
+            // Use local scraper
+            AddonPtr addon{};
+            CServiceBroker::GetAddonMgr().GetAddon("metadata.local", addon,
+                                                   OnlyEnabled::CHOICE_YES);
+            scraper = std::dynamic_pointer_cast<CScraper>(addon);
+
+            // See if Movie NFO
+            movie->SetPath(movie->GetDynPath());
+            const auto nfo{std::make_unique<CVideoTagLoaderNFO>(*movie, scraper, true)};
+            if (nfo->HasInfo())
+            {
+              // Movie NFO found - see if it has set overview
+              nfo->Load(tag, false);
+
+              // Check set matches
+              if (tag.m_set.GetTitle() == originalSetTitle)
+              {
+                localFound = true;
+                if (tag.m_set.HasOverview())
+                  overview = tag.m_set.GetOverview();
+              }
+            }
+
+            // Get online information (only if nothing found locally)
+            if (!onlineFound && overview.empty() && !movie->GetVideoInfoTag()->GetTitle().empty())
+            {
+              scraper = db.GetScraperForPath(movie->GetDynPath(), scanSettings);
+              CVideoInfoDownloader infoDownloader{scraper};
+              MOVIELIST itemResultList;
+              infoDownloader.FindMovie(movie->GetVideoInfoTag()->GetTitle(),
+                                       movie->GetVideoInfoTag()->GetYear(), itemResultList);
+              if (!itemResultList.empty())
+              {
+                std::unordered_map<std::string, std::string> uniqueIDs;
+                if (infoDownloader.GetDetails(uniqueIDs, itemResultList.at(0), tag))
+                  if (tag.m_set.HasTitle())
+                  {
+                    onlineFound = true;
+                    if (overview.empty())
+                      overview = tag.m_set.GetOverview();
+                  }
+              }
+            }
+          }
+        }
+      }
+
+      // Use Set.NFO, if present, as priority
+      bool setFound{CVideoInfoScanner::UpdateSetInTag(tag)};
+      if (tag.m_set.HasTitle())
+        overview = tag.m_set.GetOverview();
+
+      // Nothing found to update with
+      if (!localFound && !onlineFound && !setFound)
+        return false;
+
+      // tag now contains up-to-date set title
+      db.AddSet(tag.m_set.GetTitle(), overview, tag.m_set.GetOriginalTitle());
+
+      // Now deal with art
+      // Clear art first
+      CTextureDatabase textureDb;
+      if (textureDb.Open())
+      {
+        for (const auto& artwork : m_item->GetArt())
+          textureDb.InvalidateCachedTexture(artwork.second);
+        textureDb.Close();
+      }
+      m_item->ClearArt();
+
+      // If poster specified in set.nfo use that first
+      CGUIListItem::ArtMap movieSetArt;
+      if (tag.m_set.HasPoster())
+      {
+        movieSetArt.insert({"poster", tag.m_set.GetPoster()});
+      }
+      else
```

Comment:
This `else` seems to prevent loading any local artwork if NFO has a poster / 'thumb'.

Reply:
See above